### PR TITLE
Add marketplace image to mariner

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

This adds a marketplace image to mariner repo. The next step is to configure the pipelines to publish these images on a regular basis.

###### Change Log  
- azurevm-packages.json contains the required cloud specific packages.
- cloud-init.cfg contains the cloud specific config, can be merged with cloud-mariner.cfg in 2.0
- cloud-init.cfg was modified to stop the waagent on hyper-v VMs.

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Locally built image. Tested on Azure gen1 Vm, Azure gen2 VM and hyper-v gen1 VM.

### CODE REVIEW

File: toolkit/imageconfigs/additionalconfigs/cloud-init.cfg [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+ - scripts-vendor
+ - scripts-per-once
+ - scripts-per-boot
+ - scripts-per-instance
+ - scripts-user
+ - ssh-authkey-fingerprints
+# - keys-to-console
+# - phone-home
+ - final-message
+# - power-state-change
```
Comment (hbeberman) [inline toolkit/imageconfigs/additionalconfigs/cloud-init.cfg:98]: Clean up the commented out options in this section and the ones above it.

File: toolkit/imageconfigs/packagelists/azurevm-packages.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+    "packages": [
+        "cloud-init",
+        "cloud-utils-growpart",
+        "dhcp-client",
+        "hyperv-daemons",
+        "openssh-server",
+        "WALinuxAgent"
+    ]
+}
```
Comment (hbeberman) [inline toolkit/imageconfigs/packagelists/azurevm-packages.json:10]: nit: Should have a newline at the end.

File: toolkit/imageconfigs/marketplace-gen2.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
+{
+    "Disks": [
+        {
+            "PartitionTableType": "gpt",
+            "MaxSize": 600,
```
Comment (hbeberman) [inline toolkit/imageconfigs/marketplace-gen2.json:5]: 600MB is fine for the initial version so we can get started on the pipeline side of things, but we should discuss how much empty space we want there to be without specifying --os-disk-size-gb

File: toolkit/imageconfigs/additionalconfigs/cloud-init.cfg [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+# This will cause the set+update hostname module to not operate (if true)
+preserve_hostname: false
+
+# datasources to check for cloud-config
+datasource_list: [ 
+                  NoCloud, 
+                  ConfigDrive,
+                  OVF,
+                  OpenStack,
```
Comment (hbeberman) [inline toolkit/imageconfigs/additionalconfigs/cloud-init.cfg:27]: Can we remove OVF and OpenStack from this list?

File: toolkit/imageconfigs/additionalconfigs/cloud-init.cfg [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+#                  OpenNebula, 
+#                  AltCloud, 
+#                  MAAS, 
+#                  GCE, 
+#                  CloudSigma, 
+#                  Ec2, 
+#                  CloudStack,
+#                  VmxGuestinfo, 
+                  None,
+                  Azure
```
Comment (hbeberman) [inline toolkit/imageconfigs/additionalconfigs/cloud-init.cfg:37]: If we bump Azure to the top of the list here will it still work in the cloud and in Hyper-V?

**Review by hbeberman — CHANGES_REQUESTED** [review]

File: toolkit/imageconfigs/additionalconfigs/configure-image.sh [hunk 6]
```diff
@@ -0,0 +1,2 @@
+echo Enabling service for Azure Serial Console
+systemctl enable serial-getty@ttyS0.service
```
[the remark this answers is not in the document]
Comment (neha170) [reply toolkit/imageconfigs/additionalconfigs/configure-image.sh:2]: Done

File: toolkit/imageconfigs/packagelists/azurevm-packages.json (hunk 2 above)
Comment (neha170) [reply toolkit/imageconfigs/packagelists/azurevm-packages.json:10]: Done

File: toolkit/imageconfigs/additionalconfigs/cloud-init.cfg (hunk 1 above)
Comment (neha170) [reply toolkit/imageconfigs/additionalconfigs/cloud-init.cfg:98]: Done

File: toolkit/imageconfigs/additionalconfigs/cloud-init.cfg (hunk 5 above)
Comment (neha170) [reply toolkit/imageconfigs/additionalconfigs/cloud-init.cfg:37]: Done

**Review by hbeberman — APPROVED** [review]
:shipit:

**Pull request merged**